### ISSUE TITLE
Allow management of individual custom URLs

### ISSUE THREAD

**Issue by mootari** [observablehq/feedback#25]
**Is your feature request related to a problem? Please describe.**
A notebook's "Set URL" option can be used repeatedly to assign multiple custom slugs to a notebook:
- If the notebook is accessed at a slug that does not match the current one, a redirect to the notebook's current URL is issued.
- A slug can only be associated with a single notebook. Previously set slugs cannot be reused, unless *all* of a notebook's slugs are reset.

When more than one slug is assigned, there is currently no way to review or remove individual slugs that have been previously set.

**Describe the solution you'd like**
1. In the "Set URL" modal, list all slugs which have been previously set for this notebook.
2. Make it possible to delete individual slugs, instead of having to reset all slugs.

**Describe alternatives you've considered**
Observable's API provides no endpoint to list all of a notebook's slugs, only the means to match / validate against existing slugs. To my knowledge there are no workarounds.

**Comment by jrus** [comment]
I believe the goal here is for people's import of a notebook (especially a pinned import) to not break because the author decided to point the URL at a different one.

**Comment by mootari** [comment]
> I believe the goal here is for cross-notebook imports (especially version-pinned imports) to not break

Yes, but you can already reset _all_ of a notebook's slugs. You just can't review or selectively remove them, hence the keyword "individual".

**Comment by jrus** [comment]
> you can already reset all of a notebook's slugs

Arguably this should be disallowed then?

**Comment by mbostock** [comment]
I’ve filed issue #34 to track version pinning of imports. My hope is that we have version pinning to prevent imports from breaking, but also we allow authors to have full control over their URLs.

**Comment by Martien** [comment]
I ran in to a related issue yesterday https://talk.observablehq.com/t/recovering-set-url-from-oblivion
